### PR TITLE
add my projects section

### PR DESCRIPTION
**Pull request by FrostEvil**

### CODE REVIEW

File: src/pages/index.astro [hunk 1]
```diff
@@ -41,6 +44,166 @@ import "../index.scss";
...
+                    Get real-time updates on current conditions, hourly
+                    forecasts, and daily outlooks. Plan your day confidently
+                    with accurate weather information at your fingertips. Never
+                    be caught off guard by the elements again!
+                  </p>
+                  <p class="w-full flex justify-around absolute bottom-0">
+                    <span class="text-orange-600">HTML</span>
+                    <span class="text-blue-600">CSS</span>
+                    <span class="text-yellow-300">JavaScript</span>
+                  </p>
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:102]: Semantically this is a list

File: src/pages/index.astro [hunk 2]
```diff
@@ -41,6 +44,166 @@ import "../index.scss";
...
+                </div>
+              </div>
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:104]: This should be extracted to a `Project` component in the `src/components` directory

**Review by aleksanderwalczuk — CHANGES_REQUESTED** [review]
Not bad. We've got the boilerplate, time to improve it 😁

File: src/pages/index.astro (hunk 1 above)
Comment (FrostEvil) [reply src/pages/index.astro]: Changed to list.

File: src/pages/index.astro (hunk 2 above)
Comment (FrostEvil) [reply src/pages/index.astro]: Added componend.

File: src/components/cardProject.astro [hunk 3]
```diff
@@ -0,0 +1,50 @@
+---
+interface Props {
+  img: string;
+  title: string;
+  description: string;
+  skills: Array<string>;
+}
```
Comment (aleksanderwalczuk) [inline src/components/cardProject.astro:7]: Now you're writing typescript, I'm amazed 🥹

File: src/components/cardProject.astro [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+const { img, title, description, skills } = Astro.props;
+---
+
+<div class="p-6 flex flex-col items-center gap-x-6 shadow-md">
+  <div class="h-64">
+    <img
+      class="rounded-3xl w-64 h-64"
+      src={img}
+      alt="Location-based weather API"
```
Comment (aleksanderwalczuk) [inline src/components/cardProject.astro:17]: This should be a prop

File: src/components/cardProject.astro [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            case "CSS":
+              return <li class="text-blue-600">{skill}</li>;
+              break;
+            case "JavaScript":
+              return <li class="text-yellow-300">{skill}</li>;
+              break;
+            default:
+              return <li class="text-cyan-500">{skill}</li>;
+              break;
+          }
```
Comment (aleksanderwalczuk) [inline src/components/cardProject.astro:45]: Try using object instead of switch

**Review by aleksanderwalczuk — CHANGES_REQUESTED** [review]
Great progress!

File: src/components/cardProject.astro (hunk 5 above)
Comment (FrostEvil) [reply src/components/cardProject.astro]: Changed

File: src/components/cardProject.astro (hunk 4 above)
Comment (FrostEvil) [reply src/components/cardProject.astro]: Add img object into Props

File: src/components/cardProject.astro [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            CSS: <li class="text-blue-600">{skill}</li>,
+            JavaScript: <li class="text-yellow-600">{skill}</li>,
+          };
+
+          return object[skill] ? (
+            object[skill]
+          ) : (
+            <li class="text-cyan-600">{skill}</li>
+          );
+        })
```
[the remark this answers is not in the document]
Comment (aleksanderwalczuk) [reply src/components/cardProject.astro:48]: beside that you can assign keys as skills
```ts
const skillToCss = {
  html: "text-orange-600",
  css: "..."
}
```

With this you can do following: `<li class={`text-${skillToCss[skill]}-600`}>{skill}</li>`

File: src/components/cardProject.astro
Comment (aleksanderwalczuk) [inline src/components/cardProject.astro:1]: component filename should be capitalized

File: src/components/cardProject.astro [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  skills: Array<string>;
+}
+
+const {
+  img: { classProp, src, alt },
+  title,
+  description,
+  skills,
+} = Astro.props;
+---
```
Comment (aleksanderwalczuk) [inline src/components/cardProject.astro:19]: between `---` you can write js/ts which will be used below in the template (check astro docs)

**Review by aleksanderwalczuk — CHANGES_REQUESTED** [review]
Few more fixes

File: src/components/cardProject.astro
Comment (aleksanderwalczuk) [reply src/components/cardProject.astro:1]: start also with capital letter, should be `CardProject.astro`

File: src/pages/index.astro [hunk 8]
```diff
@@ -41,6 +45,86 @@ import "../index.scss";
...
+                img={{
+                  classProp: "rounded-3xl w-64 h-64",
+                  src: "astro-lp/project_4.png",
+                  alt: "Forest Group welcome page",
+                }}
+                title="Forest Group"
+                description=`Welcome to Forest Group: Your digital oasis for landscaped parks. Explore lush green spaces, tranquil gardens, and vibrant landscapes from the comfort of your screen. Discover inspiration for your next outdoor adventure, family picnic, or leisurely stroll.`
+                skills={["HTML", "CSS", "JavaScript"]}
+              />
+            </li>
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:123]: This is not dry, use `.map` to fix it :)

Comment (FrostEvil) [reply src/pages/index.astro]: Changed

File: src/components/CardProject.astro [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    </p>
+    <ul class="w-full flex justify-around absolute bottom-0">
+      {
+        skills.map((skill) => {
+          return skillToCSS[skill] ? (
+            <li class={skillToCSS[skill]}>{skill}</li>
+          ) : (
+            <li class="text-cyan-600">{skill}</li>
+          );
+        })
```
Comment (aleksanderwalczuk) [inline src/components/CardProject.astro:48]: Nice, using arrow fn we can omit return like this:
```jsx
        skills.map((skill) => ( skillToCSS[skill] ? (
            <li class={skillToCSS[skill]}>{skill}</li>
          ) : (
            <li class="text-cyan-600">{skill}</li>
          );
        ))
```

File: src/pages/index.astro [hunk 10]
```diff
@@ -1,10 +1,54 @@
 ---
 import Layout from "../layouts/Layout.astro";
 import "../index.scss";
+import CardProject from "../components/CardProject.astro";
+
+const weatherCard = {
+  imgSrc: "astro-lp/project_1.png",
+  imgAlt: "Weather App Layout",
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:8]:
```tsx
{
 img: {
   src: "",
   alt: ""
   }
//...
}

File: src/pages/index.astro [hunk 11]
```diff
@@ -41,6 +85,56 @@ import "../index.scss";
...
+          inspiration, marketing strategies, or technical solutions, my
+          portfolio highlights demonstrate my ability to tackle diverse
+          challenges and deliver outstanding outcomes. Dive in and discover the
+          depth of my professional experience and the breadth of my creative
+          vision.
+        </p>
+        <div class="mt-24">
+          <ul class="grid md:grid-cols-2 gap-8 md:gap-16">
+            {
+              cardObjects.map((card) => {
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:116]:
```suggestion
              cards.map((card) => {
```

File: src/pages/index.astro [hunk 12]
```diff
@@ -41,6 +85,56 @@ import "../index.scss";
...
+                return (
+                  <li class="flex justify-stretch">
+                    <CardProject
+                      img={{
+                        classProp: "rounded-3xl w-64 h-64",
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:121]: If those classes are the same for each card we don't need this prop

File: src/components/CardProject.astro (hunk 9 above)
Comment (FrostEvil) [reply src/components/CardProject.astro]: Changed

File: src/pages/index.astro (hunk 10 above)
Comment (FrostEvil) [reply src/pages/index.astro]: Changed

File: src/pages/index.astro (hunk 11 above)
Comment (FrostEvil) [reply src/pages/index.astro]: Changed

File: src/pages/index.astro (hunk 12 above)
Comment (FrostEvil) [reply src/pages/index.astro]: Changed

File: src/pages/index.astro [hunk 13]
```diff
@@ -113,22 +120,16 @@ const cardObjects = [weatherCard, calculatorCard, realEstateCard, forestCard];
...
-                );
-              })
+              cards.map((card) => (
+                <li class="flex justify-stretch">
+                  <CardProject
+                    img={card.img}
+                    title={card.title}
+                    description={card.description}
+                    skills={card.skills}
+                  />
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:130]: Much better, one last request to move on. All these props come from the same source - let's adjust the Card props to accept just `card` containing all existing card data

File: src/pages/index.astro [hunk 14]
```diff
@@ -4,44 +4,51 @@ import "../index.scss";
...
-  imgSrc: "astro-lp/project_1.png",
-  imgAlt: "Weather App Layout",
+  img: {
+    src: "astro-lp/project_1.png",
+    alt: "Weather App Layout",
+  },
   title: "Weather App",
   description:
     "Stay ahead of the forecast with our intuitive weather app. Get real-time updates on current conditions, hourly forecasts, and daily outlooks. Plan your day confidently with accurate weather information at your fingertips. Never be caught off guard by the elements again!",
   skills: ["HTML", "CSS", "JavaScript"],
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:14]: This value is the same for every card, lets move it to `const baseSkills = ["HTML", "CSS", "JavaScript"];` and assign this const to each obj

File: src/components/CardProject.astro [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+const {
+  img: { src, alt },
+  title,
+  description,
+  skills,
+} = Astro.props;
+
+const skillToCSS = {
```
Comment (aleksanderwalczuk) [inline src/components/CardProject.astro:19]: let's rename it to `mapSkillToCss`

File: src/pages/index.astro [hunk 16]
```diff
@@ -41,6 +92,49 @@ import "../index.scss";
...
+          inspiration, marketing strategies, or technical solutions, my
+          portfolio highlights demonstrate my ability to tackle diverse
+          challenges and deliver outstanding outcomes. Dive in and discover the
+          depth of my professional experience and the breadth of my creative
+          vision.
+        </p>
+        <div class="mt-24">
+          <ul class="grid md:grid-cols-2 gap-8 md:gap-16">
+            {
+              cards.map((card) => (
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:123]: use destructuring
```suggestion
              cards.map(({title, description, img, skills}) => (
```

**Comment by aleksanderwalczuk** [comment]
Mark this PR as ready to review when you push last changes

File: src/pages/index.astro (hunk 13 above)
Comment (FrostEvil) [reply src/pages/index.astro]: Changed

File: src/pages/index.astro (hunk 14 above)
Comment (FrostEvil) [reply src/pages/index.astro]: Changed

File: src/components/CardProject.astro (hunk 15 above)
Comment (FrostEvil) [reply src/components/CardProject.astro]: Changed

File: src/pages/index.astro (hunk 16 above)
Comment (FrostEvil) [reply src/pages/index.astro:125]: destructuring done in the card component

File: src/components/CardProject.astro [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+const { card } = Astro.props;
+
+const {
+  img: { src, alt },
+  title,
+  description,
+  skills,
+} = card;
```
Comment (aleksanderwalczuk) [inline src/components/CardProject.astro:21]: use nested destruct instead

Comment (aleksanderwalczuk) [reply src/components/CardProject.astro:21]: (2 lvls deep is still readable)

File: src/pages/index.astro [hunk 18]
```diff
@@ -1,10 +1,63 @@
 ---
 import Layout from "../layouts/Layout.astro";
 import "../index.scss";
+import CardProject from "../components/CardProject.astro";
+
+const baseSkills = ["HTML", "CSS", "JavaScript"];
+
+const weatherCard = {
+  img: {
+    src: "astro-lp/project_1.png",
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:10]: not sure if it's good to keep base path hardcoded, WDYT?

File: src/components/CardProject.astro [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+      alt: string;
+    };
+    title: string;
+    description: string;
+    skills: Array<string>;
+  };
+}
+
+const {
+  card,
```
Comment (aleksanderwalczuk) [inline src/components/CardProject.astro:15]: No need for this line

**Review by aleksanderwalczuk — CHANGES_REQUESTED** [review]
Small adjustment and we're good to go

File: src/components/CardProject.astro (hunk 19 above)
Comment (FrostEvil) [reply src/components/CardProject.astro:15]: Deleted

**Review by aleksanderwalczuk — APPROVED** [review]

**Pull request merged**